e2e/benchmark_test.go

```diff
@@ -8,35 +8,47 @@ import (
 	"testing"
 
 	"github.com/Layr-Labs/eigenda-proxy/clients/standard_client"
+	"github.com/Layr-Labs/eigenda-proxy/testutils"
+	"github.com/Layr-Labs/eigenda-proxy/testutils/testmatrix"
 )
 
 // BenchmarkPutsWithSecondary  ... Takes in an async worker count and profiles blob insertions using
 // constant blob sizes in parallel
-// TODO: test variations with V2
 func BenchmarkPutsWithSecondary(b *testing.B) {
-	testCfg := TestConfig(true, false)
-	testCfg.UseS3Caching = true
-	writeThreadCount := os.Getenv("WRITE_THREAD_COUNT")
-	threadInt, err := strconv.Atoi(writeThreadCount)
-	if err != nil {
-		panic(fmt.Errorf("Could not parse WRITE_THREAD_COUNT field %w", err))
-	}
-	testCfg.WriteThreadCount = threadInt
+	testMatrix := testmatrix.NewTestMatrix()
+	testMatrix.AddDimension(testmatrix.NewDimension(testutils.V2Enabled, []any{true, false}))
+	testMatrix.AddDimension(testmatrix.NewDimension(testutils.Backend, []any{testutils.Memstore}))
```

Comment:
this feels overly verbose and more error prone than what we had before since we're introducing the iteration within each test case - is it not possible to capture the matrix in a way that doesn't require introducing a table driven test over each e2e test?

Reply:
Based on your feedback, I went back to the drawing board and made things much simpler. LMK what you think